csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/CXX_Api_Sample.cpp

```diff
@@ -126,6 +126,9 @@ int main(int argc, char* argv[]) {
   // Score for class[2] = 0.000125
   // Score for class[3] = 0.001180
   // Score for class[4] = 0.001317
+
+  for (size_t i = 0; i < num_input_nodes; i++)
+    allocator.Free(static_cast<void*>(const_cast<char*>(input_node_names[i])));
   printf("Done!\n");
```

Comment:
Would be better to make input_node_names contain non-const char*s than to do the const_cast here.

Reply:
> Using 'const char*' in inputs_names[] and output_names[] however doesn't need to change. 
In onnxruntime/test/opaque_api/test_opaque_api.cc, better to using const char*, why non-const char*s are better here.

Reply:
The call to GetInputName/GetOutputName returns a non-const pointer to memory that is now owned by the calling code. Storing that in a const char* essentially hides the fact it owns the memory as typically if you're given something that is const, you do not own it, and you should not change/free it.

In test_opaque_api.cc there are individual variables storing the values returned by GetInputName/GetOutputName prior to them being added to an array for the call to Run. Those individual variables can be used in the call to Free.

I think there should be something equivalent here, with something storing the non-const char*'s returned by GetInputName so those can be used in the call to allocator.Free without a const_cast. That way the ownership semantics are clearer, and the ugly const_cast can be avoided.

Reply:
clear explanation, I got it.